docs/sources/k6/next/misc/versioning-and-stability-guarantees.md

```diff
@@ -0,0 +1,265 @@
+
+### JavaScript interpreter
+
+With k6 scripts written in JavaScript and the environment in which they are executed supporting a specific set of JavaScript syntax and ECMAScript features, the k6 JavaScript interpreter is considered part of the k6 stable and supported API surface.
+
+This ensures that users can write and execute scripts with predictable behavior and compatibility over the lifespan of their major version of k6.
+### Modules
+
+Furthermore, the *“core”* modules exposed directly to a script by k6 *(e.g.,* `k6`, `k6/http`, `k6/execution`) are another critical aspect of the k6 experience and of the interface between k6 and its users and is as such part of the stable and supported API surface.
+It includes any class, properties, methods, function, or relative signature that is part of the module and is publicly exposed and documented.
+
+Here, the full list of the `core` modules: `k6`, `k6/browser`, `k6/crypto`, `k6/crypto/x509`, `k6/data`, `k6/encoding`, `k6/execution`,  `k6/http`, `k6/metrics`, `k6/net/grpc`, `k6/timers`, `k6/ws`.
+
+Experimental modules recognizable from their `k6/experimental` prefixed import path are explicitly excluded from the stable and supported API surface.
+### Extensions
+
+Our stable and supported API surface also includes the extensions API, which empowers internal and external developers to build custom modules and outputs as part of the k6 codebase or as external extensions.
+
+This includes the [go.k6.io/k6/js/modules](https://pkg.go.dev/go.k6.io/k6@v0.57.0/js/modules) Go package that internal and external developers have available to build module extensions, as well as the [go.k6.io/k6/output](https://pkg.go.dev/go.k6.io/k6@v0.57.0/output) that can be used to build custom outputs for k6.
+
+It also includes public k6 Go packages that are exposed to developers to facilitate the development of these extensions (e.g., the current  [go.k6.io/k6/lib.State](https://pkg.go.dev/go.k6.io/k6@v0.57.0/lib#State) package).
+### Output
+
+We guarantee the stability and support of metrics, logs, traces, and profiling outputs that we fully control in an end-to-end flow. This includes the JSON, CSV, JSON summary exports, cloud metrics, logs and traces outputs, and OpenTelemetry formats while excluding outputs dependent on third-party platforms and protocols.
+### Metrics
+
+We guarantee the stability of metric names, types, and units. Any changes to these are considered breaking and will be handled according to our [versioning strategy](#versioning-strategy). Doing so ensures that any product or artifact (dashboards, etc.) built on top of k6 can maintain a reliable and consistent experience.
+## Deprecation, breaking changes, and migration policy
+
+### Breaking Changes
+
+In a constantly evolving product, [breaking changes](#breaking-change) are sometimes unavoidable. They require that k6 users have to actively react to them because the state after it is not backward compatible with the state before.
```

Comment:
```suggestion
In a constantly evolving product, [breaking changes](#breaking-change) are sometimes unavoidable. They require that k6 users actively react to them because the state after the changes is not backward compatible with the previous state.
```